dns_discover.go

```diff
@@ -4,18 +4,25 @@ package fargo
 
 import (
 	"fmt"
+	"time"
+
 	"github.com/cenkalti/backoff"
 	"github.com/franela/goreq"
 	"github.com/miekg/dns"
-	"time"
 )
 
 const azURL = "http://169.254.169.254/latest/meta-data/placement/availability-zone"
 
 var ErrNotInAWS = fmt.Errorf("Not in AWS")
 
-func discoverDNS(domain string, port int, urlBase string) (servers []string, ttl time.Duration, err error) {
-	r, _ := region()
+func discoverDNS(domain string, port int, urlBase string, clientRegion string) (servers []string, ttl time.Duration, err error) {
+	var r string
+	if clientRegion != "" {
+		r = clientRegion
+	} else {
+		regionResolvedAutomatically, _ := region()
+		r = regionResolvedAutomatically
```

Comment:
Why are there two ways to set the region? Why not always resolve automatically if it is possible?

Reply:
The old code was always trying to resolve this automatically, yes, using AWS API. It would see that a service instance is in europe, and try to resolve from there.

Unfortunately, the way eureka is configured at Coveo, we have to "force" it to use the main region always (us-east).